### PR TITLE
Update maven to version 3.8.7

### PR DESCRIPTION
**Pull request by Sumynwa**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [ ] Ready to merge

---

###### Summary 
Update package to version 3.8.7

###### Change Log  
- lint spec
- minor fixes in cache generation script

###### Does this affect the toolchain?  
NO

###### Test Methodology

- Pipeline build id: Buddy Build
                              [AMD64](https://dev.azure.com/mariner-org/mariner/_build/results?buildId=309957)
                              [ARM64](https://dev.azure.com/mariner-org/mariner/_build/results?buildId=309902)
                              [**NOTE**: Reason for both pipeline run's being marked as 'Fail' is due to missing Requires package msopenjdk-17]

### CODE REVIEW

**Review by mfrw — APPROVED** [review]

File: SPECS/maven/maven.spec [hunk 1]
```diff
@@ -73,7 +66,7 @@ popd
 # Changing distribution dir to BUILD directory as install macro clears buildroot prior to creating a fresh directory, thus clearing artifacts copied by maven. We copy them later.
 MAVEN_DIST_DIR=%{_builddir}%{_prefixmvn}
 
-export JAVA_HOME="%{_libdir}/jvm/msopenjdk-11"
+export JAVA_HOME=$(find %{_libdir}/jvm -name "msopenjdk*")
```
Comment (jslobodzian) [inline SPECS/maven/maven.spec:69]: Do we want to keep this change?  Seems like the explicit -11 was/is correct?

File: SPECS/maven/maven.spec [hunk 2]
```diff
@@ -133,6 +126,10 @@ cp %{_builddir}/apache-maven-%{version}/apache-maven/README.txt %{buildroot}%{_p
 %{_prefixmvn}/README.txt
 
 %changelog
+* Wed Feb 15 2023 Sumedh Sharma <sumsharma@microsoft.com> - 3.8.7-1
+- Update to version 3.8.7
+- Minor fixes in maven_build_caches script
```
Comment (jslobodzian) [inline SPECS/maven/maven.spec:131]: Please be more explicit about what the minor fixes were

File: SPECS/maven/maven.spec (hunk 1 above)
Comment (Sumynwa) [reply SPECS/maven/maven.spec:69]: Yes, JAVA_HOME should be explicitly set to point to msopenjdk-11 during build. I will upate.

Comment (Sumynwa) [reply SPECS/maven/maven.spec]: Done.

File: SPECS/maven/maven.spec (hunk 2 above)
Comment (Sumynwa) [reply SPECS/maven/maven.spec]: Removed.

**Pull request closed without merging**